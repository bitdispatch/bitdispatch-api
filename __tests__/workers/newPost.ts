import { Connection, getConnection } from 'typeorm';

import { expectSuccessfulBackground } from '../helpers';
import worker from '../../src/workers/newPost';
import { Post, Source } from '../../src/entity';

let con: Connection;

beforeAll(async () => {
  con = await getConnection();
});

it('should save a new post', async () => {
  await con.getRepository(Source).save([
    {
      id: 'bitcoinist',
      twitter: 'bitcoinist',
      website: 'https://bitcoinist.com',
      active: true,
      rankBoost: 22,
      name: 'Bitcoinist',
      image:
        'https://bitcoinist.com/wp-content/uploads/2021/04/cropped-cropped-cropped-Icon-192x192.png',
      private: false,
    },
  ]);

  await expectSuccessfulBackground(worker, {
    id: '185976c940eb141db4cfffb9c0547875',
    title:
      'Former Korean Finance Minister Joins Hashed Open Research to Promote Blockchain',
    tags: ['industry'],
    publishedAt: '2022-08-28T12:02:19.000Z',
    publicationId: 'bitcoinist',
    url: 'https://bitcoinist.com/former-korean-finance-minister-joins-hashed-open-research-to-promote-blockchain/',
    image:
      'https://res.cloudinary.com/dysahequr/image/upload/f_auto,q_auto/v1/posts/781781251e78a42c94a44dd4e86b9807',
    siteTwitter: '@bitcoinist',
    creatorTwitter: '@bitcoinist',
    readTime: 1,
    canonicalUrl:
      'https://bitcoinist.com/former-korean-finance-minister-joins-hashed-open-research-to-promote-blockchain/',
    placeholder:
      'data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAAFAAoDASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAAAAUH/8QAHhAAAgIDAQADAAAAAAAAAAAAAQIDBQAEEQYSMkH/xAAUAQEAAAAAAAAAAAAAAAAAAAAD/8QAFxEAAwEAAAAAAAAAAAAAAAAAAAEiQv/aAAwDAQACEQMRAD8Ayny89XYelqdSWoT4SbUayA7Dsrr3hHD+HI1rGkdpuJECkazOqqD9QGPBjGOlQeT/2Q==',
    ratio: 1.75,
  });
  const posts = await con.getRepository(Post).find();
  expect(posts.length).toEqual(1);
  expect(posts[0]).toMatchSnapshot();
});
