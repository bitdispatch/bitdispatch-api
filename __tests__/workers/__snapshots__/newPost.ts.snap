// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should save a new post 1`] = `
Post {
  "authorId": null,
  "banned": false,
  "canonicalUrl": "https://bitcoinist.com/former-korean-finance-minister-joins-hashed-open-research-to-promote-blockchain/",
  "comments": 0,
  "createdAt": 2022-09-19T17:01:42.968Z,
  "creatorTwitter": "@bitcoinist",
  "deleted": false,
  "description": null,
  "discussionScore": null,
  "id": "WoDVfnI9X",
  "image": "https://res.cloudinary.com/dysahequr/image/upload/f_auto,q_auto/v1/posts/781781251e78a42c94a44dd4e86b9807",
  "lastTrending": null,
  "metadataChangedAt": 2022-09-19T17:01:42.995Z,
  "placeholder": "data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAAFAAoDASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAAAAUH/8QAHhAAAgIDAQADAAAAAAAAAAAAAQIDBQAEEQYSMkH/xAAUAQEAAAAAAAAAAAAAAAAAAAAD/8QAFxEAAwEAAAAAAAAAAAAAAAAAAAEiQv/aAAwDAQACEQMRAD8Ayny89XYelqdSWoT4SbUayA7Dsrr3hHD+HI1rGkdpuJECkazOqqD9QGPBjGOlQeT/2Q==",
  "publishedAt": 2022-08-28T12:02:19.000Z,
  "ratio": 1.75,
  "readTime": 1,
  "score": 27726781,
  "scoutId": null,
  "sentAnalyticsReport": true,
  "shortId": "WoDVfnI9X",
  "siteTwitter": "@bitcoinist",
  "sourceId": "bitcoinist",
  "summary": null,
  "tagsStr": null,
  "title": "Former Korean Finance Minister Joins Hashed Open Research to Promote Blockchain",
  "toc": null,
  "trending": null,
  "tweeted": false,
  "upvotes": 0,
  "url": "https://bitcoinist.com/former-korean-finance-minister-joins-hashed-open-research-to-promote-blockchain/",
  "views": 0,
  "viewsThreshold": 0,
}
`;
